scrape_data/scrape_schedule_versions.py

```diff
@@ -0,0 +1,209 @@
+import pendulum
+import logging
+import calendar
+import pandas as pd
+
+logger = logging.getLogger()
+logging.basicConfig(level=logging.INFO)
+logger.setLevel(logging.INFO)
+
+BASE_URL = "https://transitfeeds.com"
+
+
+def check_latest_rt_data_date() -> str:
+    """Fetch the latest available date of real-time bus data
+
+    Returns:
+        str: A string of the latest date in YYYY-MM-DD format.
+    """
+    if pendulum.now("America/Chicago").hour >= 11:
+        end_date = pendulum.yesterday("America/Chicago").date().format("YYYY-MM-DD")
+    else:
+        end_date = (
+            pendulum.now("America/Chicago").subtract(days=2).date().format("YYYY-MM-DD")
+        )
+    return end_date
+
+
+def fetch_schedule_versions(month: int, year: int) -> List[pendulum.date]:
+    """Get the schedule versions from transitfeeds.com from the most recent
+       to specified month and year (inclusive). In case there are
+       multiple schedules for a given month and year pair,
+       all schedules will be fetched.
+
+    Args:
+        month (int): The month of interest
+        year (int): The year of interest
+
+    Returns:
+        List[pendulum.date]: A list of unique schedule versions
+    """
+    link_list = []
+    page = 1
+    found = False
+    while not found:
+        logging.info(f" Searching page {page}")
+        url = BASE_URL + f"/p/chicago-transit-authority/165?p={page}"
+        response = requests.get(url).content
+        soup = BeautifulSoup(response, "lxml")
+        # List of dates from first row
+        table = soup.find_all("table")
+        for row in table[0].tbody.find_all("tr"):
+            first_col = row.find_all("td")[0]
+            date = pendulum.parse(first_col.text.strip(), strict=False)
+            # Find schedules up to and including the specified date.
+            if date.month == month and date.year == year:
+                logging.info(
+                    f" Found schedule for"
+                    f" {calendar.month_name[date.month]} {date.year}"
+                )
+                logging.info(
+                    f" Adding schedule for {calendar.month_name[date.month]}"
+                    f" {date.day}, {date.year}"
+                )
+                link_list.append(first_col)
+                found = True
+                continue
+            if found:
+                break
+            link_list.append(first_col)
+        page += 1
+
+    date_list = [s.text.strip() for s in link_list]
+    # Check for duplicates. The presence of duplicates could mean
+    # that the schedule was not in-effect.
+    # See https://github.com/chihacknight/chn-ghost-buses/issues/30
+    duplicates = pd.Series(date_list)[pd.Series(date_list).duplicated()].values
+    if len(duplicates) > 0:
+        logging.info(
+            f" The duplicate schedule versions are"
+            f" {set(duplicates)}. Check whether these were in-effect."
+        )
+        # Keep the first occurrence of duplicates.
```

Comment:
Just checking here - "first" means "the one that appears first in the order of display on the Transitfeeds site, which was actually scraped later"? (That would be the correct implementation IMO, just wasn't sure whether "first" means chronologically or in display order which is reverse chronological)

Reply:
Yes "first" here means the first in the order of display on Transitfeeds. 

`date_list` appears as 
```python
[...'14 September 2021', '7 September 2021', '1 September 2021',
  '1 September 2021', '1 September 2021', '1 September 2021',
  '1 September 2021', '2 August 2021', '15 June 2021',..]
```
So taking the first will take the latest version. I've added a note on that in the comments.